src/Arcus.Security.Core/Caching/CachedSecretProvider.cs

```diff
@@ -129,5 +130,16 @@ public async Task<Secret> GetSecretAsync(string secretName, bool ignoreCache)
 
             return secret;
         }
+
+        /// <summary>
+        /// Removes the secret with the given <paramref name="secretName"/> from the cache;
+        /// so the next time <see cref="GetSecretAsync(string)"/> is called, a new version of the secret will be added back to the cache.
+        /// </summary>
+        /// <param name="secretName">The name of the secret that should be removed from the cache.</param>
+        public Task InvalidateSecretAsync(string secretName)
+        {
+            _memoryCache.Remove(secretName);
```

Comment:
What if it's not there? Does it throw? Should we check?

Reply:
`null` will be returned for `MemoryCache`; but it's an interface we take in the ctor; so could be anything.

Reply:
Hm there is no way to verify if it's there so it's ok.